networkd/network_server.ml

```diff
@@ -521,10 +533,14 @@ module Interface = struct
                                      ethtool_settings; ethtool_offload; _} as c)) ->
             update_config name c;
             exec (fun () ->
-                (* We only apply the DNS settings when in static IPv4 mode to avoid conflicts with DHCP mode.
+                (* We only apply the DNS settings when not in a DHCP mode to avoid conflicts.
                    					 * The `dns` field should really be an option type so that we don't have to derive the intention
                    					 * of the caller by looking at other fields. *)
-                match ipv4_conf with Static4 _ -> set_dns () dbg ~name ~nameservers ~domains | _ -> ());
+                match (ipv4_conf, ipv6_conf) with
+                  | (Static4 _, _)
+                  | (_, Static6 _)
+                  | (_, Autoconf6) -> set_dns () dbg ~name ~nameservers ~domains
+                  | _ -> ());
```

Comment:
This fix is available alone in a new PR (https://github.com/xapi-project/xcp-networkd/pull/181) since the rest of this one is being discussed.